### PR TITLE
feat(av-cliper): MP4Clip thumbnails support geting by interval

### PR DESCRIPTION
**Pull request by izkizk8**
This pull request primarily modifies the `thumbnails` method in the `MP4Clip` class in the `packages/av-cliper/src/clips/mp4-clip.ts` file, and updates the corresponding test in the `packages/av-cliper/src/clips/__tests__/mp4-clip.test.ts` file. The `thumbnails` method now accepts additional parameters for start time, end time, and interval time in microseconds, and the method's behavior has been updated to generate thumbnails at these specified intervals. The test has been updated to reflect these changes.

Changes to the `MP4Clip` class:

* [`packages/av-cliper/src/clips/mp4-clip.ts`](diffhunk://#diff-d84eb28cb220fd605837400024ab399e668c9c0f21243f414431c8fa0c83e683L245-R259): The `thumbnails` method now accepts additional parameters `start`, `end`, and `interval` for specifying the start time, end time, and interval time in microseconds. If the `interval` parameter is provided, the method generates thumbnails at the specified intervals. If not, it generates thumbnails at keyframes. A helper function `pushPngPromise` has been added to push the generated thumbnail to the `pngPromises` array. The filter in the `samples` constant has been updated to also check if the sample's `cts` is within the specified start and end times. [[1]](diffhunk://#diff-d84eb28cb220fd605837400024ab399e668c9c0f21243f414431c8fa0c83e683L245-R259) [[2]](diffhunk://#diff-d84eb28cb220fd605837400024ab399e668c9c0f21243f414431c8fa0c83e683R283-R302) [[3]](diffhunk://#diff-d84eb28cb220fd605837400024ab399e668c9c0f21243f414431c8fa0c83e683L281-R313) [[4]](diffhunk://#diff-d84eb28cb220fd605837400024ab399e668c9c0f21243f414431c8fa0c83e683R323)

Changes to the test:

* [`packages/av-cliper/src/clips/__tests__/mp4-clip.test.ts`](diffhunk://#diff-ff4601bbf9909b118de7172c2c0e4399fd786269d135e6505e96edafeff1edd0L68-R69): The test for the `thumbnails` method has been updated to reflect the changes in the method. It now tests the method with the additional parameters for start time, end time, and interval time in microseconds, and expects a different number of thumbnails to be generated.

### CODE REVIEW

**Comment by hughfenghen** [comment]
@izkizk8 抱歉，我近期有点忙，很高兴你能主动实现这个功能。

提交两个 review 建议，请查看。

**Comment by izkizk8** [comment]
> @izkizk8 抱歉，我近期有点忙，很高兴你能主动实现这个功能。
> 
> 提交两个 review 建议，请查看。

@hughfenghen 感谢，不过我这边好像没有看到 review 建议，能否再 check 一下？

**Comment by hughfenghen** [comment]
忘记提交了  [doge]

File: packages/av-cliper/src/clips/__tests__/mp4-clip.test.ts [hunk 1]
```diff
@@ -65,8 +65,12 @@ test('thumbnails', async () => {
   const thumbnails = await clip.thumbnails();
   expect(thumbnails.length).toBe(9);
   expect((await createImageBitmap(thumbnails[0].img)).width).toBe(100);
-  const thumbnails150 = await clip.thumbnails(150);
-  expect(thumbnails150.length).toBe(9);
+  const thumbnails150 = await clip.thumbnails(150, {
+    start: 1e6,
+    end: 1e7,
```
Comment (hughfenghen) [inline packages/av-cliper/src/clips/__tests__/mp4-clip.test.ts:70]: 建议 `end: 10e6`， 1e6 就是 微秒 转 秒, 统一单位方便阅读。

**Comment by hughfenghen** [comment]
@izkizk8  hi， 感谢你的付出，代码很好。

如果你有空的话，可以在[视频缩略图](https://hughfenghen.github.io/WebAV/demo/6_1-video-thumbnails)补充一个 DEMO。
演示按指定步长提取帧缩略图，若你没有空我会在近期补充这个 DEMO。

**Comment by izkizk8** [comment]
> @izkizk8 hi， 感谢你的付出，代码很好。
> 
> 如果你有空的话，可以在[视频缩略图](https://hughfenghen.github.io/WebAV/demo/6_1-video-thumbnails)补充一个 DEMO。 演示按指定步长提取帧缩略图，若你没有空我会在近期补充这个 DEMO。

@hughfenghen hello, 我已经加了一个 demo，不知是否合适 。
image

**Comment by hughfenghen** [comment]
在 @webav/av-cliper@0.4.4  中发布

**Pull request merged**